src/createDynamicHostResolver.js

```diff
@@ -12,33 +12,60 @@
 
 var window = require('@adobe/reactor-window');
 
-module.exports = function (turbineEmbedCode, cdnAllowList, debugController) {
+module.exports = function (
+  turbineEmbedCode,
+  dynamicCdnEnabled,
+  cdnAllowList,
+  debugController
+) {
+  // A missing list means that we are not trying to dynamic replace (archives,
+  // sftp, no premium CDN option enabled on the company).
   // even an empty list is flagging to us that we're trying to enforce dynamic
-  var isDynamicEnforced = Array.isArray(cdnAllowList);
+  var isDynamicEnforced = Boolean(
+    dynamicCdnEnabled && Array.isArray(cdnAllowList)
+  );
   var shouldAugment = Boolean(isDynamicEnforced && turbineEmbedCode);
 
-  // TODO: web only? I think embedded TVs wouldn't have
-  //  __satellite.container.dynamicEnforced turned on
+  // using document.createElement('a') because IE10/11 doesn't support new URL()
   var turbineUrl = document.createElement('a');
-  turbineUrl.href = turbineEmbedCode;
-  if (
-    (!/^https?:\/\/.*/.test(turbineEmbedCode) || !turbineUrl.host) &&
-    isDynamicEnforced
-  ) {
-    var missingEmbedCodeError = new Error(
-      'Unable to find the Library Embed Code for Dynamic Host Resolution.'
-    );
-    missingEmbedCodeError.code = 'dynamic_host_resolver_constructor_error';
-    throw missingEmbedCodeError;
-  }
-
-  if (isDynamicEnforced && cdnAllowList.indexOf(turbineUrl.hostname) === -1) {
-    var dynamicDeniedError = new Error(
-      'This library is not authorized for this domain. ' +
-        'Please contact your CSM for more information.'
-    );
-    dynamicDeniedError.code = 'dynamic_host_not_allowed';
-    throw dynamicDeniedError;
+  if (isDynamicEnforced) {
+    // throw whenever we can't determine the embed code
+    var throwUnavailableEmbedCode = function () {
+      var missingEmbedCodeError = new Error(
+        'Unable to find the Library Embed Code for Dynamic Host Resolution.'
+      );
+      missingEmbedCodeError.code = 'dynamic_host_resolver_constructor_error';
+      throw missingEmbedCodeError;
+    };
+    if (turbineEmbedCode) {
+      var httpsMatcher = new RegExp(/^https?:\/\/.*/);
+      var doubleSlashMatcher = new RegExp(/^\/\/.*/);
+      if (
+        !httpsMatcher.test(turbineEmbedCode) &&
+        !doubleSlashMatcher.test(turbineEmbedCode)
+      ) {
+        throwUnavailableEmbedCode();
+      }
+      // try to construct the url
+      if (!httpsMatcher.test(turbineEmbedCode)) {
+        turbineUrl.href = window.location.protocol + turbineEmbedCode;
+      } else {
+        turbineUrl.href = turbineEmbedCode;
+      }
+    }
+    // check URL construction
+    if (!turbineUrl.host) {
```

Comment:
Why do you use the `host` attribute here and a few lines below you use `hostname`? What is the case when the `host` or `hostname` wouldn't be available?

Reply:
When the embed code is malformed and `turbine.href = turbineEmbedCode` does not give us a good URL. All of this would be simplified with `new URL()` if we didn't need to support IE.